src/lib.rs

```diff
@@ -520,3 +520,6 @@ pub mod doc_test {
         "guide/src/types.md" => guide_types_md,
     }
 }
+
+/// Represent [`intern!`] return type.
+pub type InternString<'py, 'a> = &'a Bound<'py, crate::types::PyString>;
```

Comment:
Typically we have written the lifetimes the other way around:

```suggestion
pub type InternString<'a, 'py> = &'a Bound<'py, crate::types::PyString>;
```